buildSrc/src/main/kotlin/source-sets-conventions.gradle.kts

```diff
@@ -154,19 +157,17 @@ kotlin {
             optIn("kotlinx.serialization.ExperimentalSerializationApi")
         }
     }
+}
 
-    targets.all {
-        compilations.all {
-            kotlinOptions {
-                if (overriddenLanguageVersion != null) {
-                    languageVersion = overriddenLanguageVersion
-                    freeCompilerArgs += "-Xsuppress-version-warnings"
-                }
-                freeCompilerArgs += "-Xexpect-actual-classes"
-            }
-        }
-        compilations["main"].kotlinOptions {
+tasks.withType(KotlinCompilationTask::class).configureEach {
+    compilerOptions {
+        if (name == "main") {
```

Comment:
`main` is the name of what?
I doubt there is a task named `main` in this project.

Reply:
It's an omission, fixed